File: 0731-my-calendar-ii/0731-my-calendar-ii.py
class MyCalendarTwo:

    def __init__(self):
        self.bookings = {}  # Dictionary to store the start and end times

    def book(self, start: int, end: int) -> bool:
        if start not in self.bookings:
            self.bookings[start] = 0
        if end not in self.bookings:
            self.bookings[end] = 0

        # Increment count for start, decrement for end
        self.bookings[start] += 1
        self.bookings[end] -= 1

        activeBookings = 0
        for time in sorted(self.bookings):
            activeBookings += self.bookings[time]
            if activeBookings >= 3:
                # Revert the changes if triple booking is detected
                self.bookings[start] -= 1
                self.bookings[end] += 1
                return False

        return True

File: 0731-my-calendar-ii/0731-my-calendar-ii.cpp
class MyCalendarTwo {
public:
    map<int,int>bookings;
    MyCalendarTwo() {

    }

    bool book(int start, int end) {
        bookings[start]++;
        bookings[end]--;
        int activeBookings = 0;
        for(const auto& [time,count] : bookings){
            activeBookings += count;
            if(activeBookings >= 3){
                bookings[start]--;
                bookings[end]++;
                return false;
            }
        }
        return true;
    }
};